core/src/test/java/org/elasticsearch/indices/cluster/IndicesClusterStateServiceRandomUpdatesTests.java

```diff
@@ -76,6 +76,15 @@
 
     private final ClusterStateChanges cluster = new ClusterStateChanges(xContentRegistry());
 
+    /**
+     * needed due to random usage of {@link IndexMetaData#INDEX_SHADOW_REPLICAS_SETTING}. removed once
+     * shadow replicas are removed.
+     */
+    @Override
+    protected boolean enableWarningsCheck() {
+        return false;
```

Comment:
Would you please add an assert guarding against this staying around after shadow replicas are removed?